src/MultiEmailInput.elm

```diff
@@ -41,15 +41,19 @@ type Msg
 type alias State =
     { nextEmail : String
     , id : String
+    , separator : Regex
+    , valid : Regex
     }
 
 
 {-| Initialize the component's state. It needs the component's ID so that it can refocus the textarea when new emails are inserted. By default, we begin with an empty textarea.
 -}
-init : String -> State
-init id =
+init : String -> Regex -> Regex -> State
+init id separator valid =
     { nextEmail = ""
     , id = id
+    , separator = separator
+    , valid = valid
```

Comment:
separator is only used on the `update` function now, whereas valid is only used on `view`. I was not that sure whether to make them a part of the state or have people supply regexes to the specific function